packages/@sanity/form-builder/src/inputs/ReferenceInput/types.ts

```diff
@@ -22,6 +22,31 @@ export interface ReferenceInfo {
   }
 }
 
+export interface ReferenceTemplate {
+  id: string
+  params?: ReferenceParams
+}
+
+export interface EditReferenceEvent {
+  id: string
+  type: string
+  template: ReferenceTemplate
+}
+
+export type ReferenceParams = Record<string, string | number | boolean>
+
+export interface CreateOption {
+  id: string
+  title: string
+  icon?: React.ReactNode
```

Comment:
I think this can also be a component type too? e.g. `icon: RocketIcon` and `icon: <RocketIcon />` would both allowed

Reply:
fixed, thanks!